src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -407,6 +407,19 @@ protected virtual void OnSaving(XmlDocument obj)
             if (handler != null) handler(obj);
         }
 
+        /// <summary>
+        /// This event is used to get the workspace model in JSON format.
+        /// This event is raised in the view model.
```

Comment:
this is not an event?

Reply:
is there a reason we need both the delegate and the event?

Reply:
because to get the return value.